benchmark/test/test_framework.py.in

```diff
@@ -164,66 +147,59 @@ def compare_output_impl(
             " ".join(["'{}'".format(arg) for arg in launcher_flags + args])
         )
     )
-    version_patterns = [
-        "    the .* module is",
+    ignore_patterns = [
+        "    the .* module is",  # version numbers
+        "DEBUG: (begin|end  ) (allocate|free)",  # allocations
```

Comment:
This is due to restricting the scope of the annotation to the repetition itself. So the new output doesn't contain allocations. But if some new implementation introduces some unwanted allocations, we would not be able to catch that.
So, I think what we want for these test is to say manually that a failed test is ok.

Reply:
This is mostly due to implementation differences in libstdc++ vector growth leading to different numbers of implementations in algorithms that use `gko::vector::push_back`